pymc_experimental/model_builder.py

```diff
@@ -187,18 +186,14 @@ def load(cls, fname):
         filepath = Path(str(fname))
         data = az.from_netcdf(filepath)
         idata = data
-        # Since there is an issue with attrs getting saved in netcdf format which will be fixed in future the following part of code is commented
-        # Link of issue -> https://github.com/arviz-devs/arviz/issues/2109
-        # if model.idata.attrs is not None:
-        #     if model.idata.attrs['id'] == self.idata.attrs['id']:
-        #         self = model
-        #         self.idata = data
-        #         return self
-        #     else:
-        #         raise ValueError(
-        #             f"The route '{file}' does not contain an inference data of the same model '{self.__name__}'"
-        #         )
-        return idata
+        if idata.attrs is not None:
+            if self.id() == idata.attrs["id"]:
+                self.idata = idata
+            else:
+                raise ValueError(
+                    f"The route '{file}' does not contain an inference data of the same model '{self.__name__}'"
+                )
+        # return idata
```

Comment:
I think we require something like 
```python
self = cls(idata.attrs["sample_config"], idata.attrs["model_config"])
self.idata = idata
return self
```

cls is the class being passed above, so we need to instantiate it using the right args from the idata.

Reply:
```
self = cls(idata.attrs["sample_config"], idata.attrs["model_config"])
self.idata = idata
```

should be in the if condition right?